akka-stream/src/main/scala/akka/stream/scaladsl/Graph.scala

```diff
@@ -534,6 +535,93 @@ final class MergeSorted[T: Ordering] extends GraphStage[FanInShape2[T, T, T]] {
   }
 }
 
+object MergeSortedN {
+  /**
+   * Create a new `MergeSortedN`.
+   */
+  def apply[A: Ordering](n: Int) = new MergeSortedN[A](n)
+}
+
+/**
+ * Merge multiple pre-sorted streams such that the resulting stream is sorted.
+ *
+ * '''Emits when''' all inputs have an element available
+ *
+ * '''Backpressures when''' downstream backpressures
+ *
+ * '''Completes when''' all upstreams complete
+ *
+ * '''Cancels when''' downstream cancels
+ */
+final class MergeSortedN[T: Ordering](inputPorts: Int) extends GraphStage[UniformFanInShape[T, T]] {
+  val inlets: immutable.IndexedSeq[Inlet[T]] = Vector.tabulate(inputPorts)(i ⇒ Inlet[T]("MergeSortedN.in" + i))
+  val out: Outlet[T] = Outlet[T]("MergeSortedN.out")
+  override val shape: UniformFanInShape[T, T] = UniformFanInShape(out, inlets: _*)
+
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = new GraphStageLogic(shape) with OutHandler {
+    private val bufferedElements = new mutable.TreeSet[(T, Int)]()
+    private val inletsBeingPulled = new mutable.HashSet[Inlet[T]]
+    private var runningInlets = inputPorts
+    private def canPush = inletsBeingPulled.isEmpty && bufferedElements.nonEmpty
+    private def upstreamsClosed = runningInlets == 0
+
+    override def preStart(): Unit = {
+      for (i ← inlets) {
+        tryPull(i)
+        inletsBeingPulled.add(i)
+      }
+    }
+
+    private def maybePush(): Unit = {
```

Comment:
I think it would be good to describe in a comment how this stage works. That it's only pulling one element from each stream and adding them to the sorted buffer. When element has been received from all streams it pushes the "smallest" element and pulls from that inlet. Waits for element from that inlet, and then again pushing the "smallest".

(if I have understood it correctly?)